x-pack/plugins/remote_clusters/public/application/sections/components/cloud_deployment_form/cloud_deployment_form.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiAccordion,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiFormRow,
+  EuiIcon,
+  EuiImage,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { useAppContext } from '../../../app_context';
+import {
+  convertCloudUrlToProxyConnection,
+  validateCloudUrl,
+} from '../../../services/cloud_deployment_url';
+
+import { cloudRemoteClustersUrl } from '../../../services/documentation';
+
+// @ts-ignore
+import Screenshot from './cloud_deployment_screenshot.png';
+
+interface Props {
+  onClose: () => void;
+  onClusterConfigure: (value: { proxyAddress: string; serverName: string }) => void;
+}
+
+// TODO copy review
+export const CloudDeploymentForm: FunctionComponent<Props> = ({ onClose, onClusterConfigure }) => {
+  const [cloudDeploymentUrl, setCloudDeploymentUrl] = useState<string>('');
+  const [error, setError] = useState<string | null>(null);
+  const { cloudDeploymentUrl: accountLink } = useAppContext();
+
+  const configureCluster = () => {
+    const urlError: string | null = validateCloudUrl(cloudDeploymentUrl);
+    if (urlError) {
+      setError(urlError);
+    } else {
+      onClose();
+      onClusterConfigure(convertCloudUrlToProxyConnection(cloudDeploymentUrl));
+    }
+  };
+  return (
+    <EuiFlyout onClose={() => onClose()}>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h2>
+            <EuiFlexGroup alignItems="center" gutterSize="s">
+              <EuiFlexItem grow={false}>
+                <EuiIcon type="logoCloud" size="l" />
+              </EuiFlexItem>
+
+              <EuiFlexItem>
+                <FormattedMessage
+                  id="xpack.remoteClusters.cloudDeploymentForm.formTitle"
+                  defaultMessage=" Add Cloud deployment as remote cluster"
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiText>
+          <p>
+            <FormattedMessage
+              id="xpack.remoteClusters.cloudDeploymentForm.formDescription"
+              defaultMessage="If you're connecting to a Cloud deployment, you can copy and paste the Elasticsearch
```

Comment:
We know this is cloud deployment (we don't support any other option) so I think the text can be more instructive.

Text suggestion: "Copy and paste the Elasticsearch endpoint URL of the remote deployment to automatically configure the remote cluster. The URL can be found on the remote deployment overview page."